### DISCUSSION TITLE
Kong max query args

### DISCUSSION THREAD

**Discussion by dbelozerovx1** [Kong/kong#10319] [Help] [answered]
Is there any way to change https://github.com/Kong/kong/blob/master/kong/pdk/request.lua#L43 for our kong? For now kong is receiving more than 100 args, but proceed exactly 100 args to upstream_uri, is there any way to turn off this limit, by far i found this https://github.com/Kong/kong/blob/master/kong/pdk/request.lua#L43 is that connected?

**Comment by StarlightIbuki** [comment]
You can call `kong.request.get_query` with an optional argument, which is the limit of the number of args.
For example, if you want to accept at maximal 200 args, you can call `kong.request.get_query(200)`.

ref: https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.request/#kongrequestget_querymax_args

**Comment by dbelozerovx1** [comment]
@StarlightIbuki yeah i've found this function, could you please clarify where and how do i call it, according to docs, that function just returns query string args, but in my case, i just want Kong to pass all the query args to upstream, which is limited by 100, do you have an example how to pass this parameter to Kong?

**Reply by StarlightIbuki** [reply]
Do you have `request-transformer` installed? I found it using this API with a hard-coded limit.
And a lot of other places are using this hard-coded limit too. We will be discussing this.

**Reply by dbelozerovx1** [reply]
yeah, we do have request-transformer

**Reply by dbelozerovx1** [reply]
do you mean without request-transformer kong will not shorten query args to 100?

**Reply by StarlightIbuki** [reply]
> do you mean without request-transformer kong will not shorten query args to 100?

I can't be sure. There're other plugins that could call the API, and some core facilities also call it. We are discussing what we could do about this.

But you could try to see if it works in your case.

**Reply by dbelozerovx1** [reply]
my appreciation, looks like plugin is limiting querystring yeah https://github.com/Kong/kong/blob/master/kong/plugins/request-transformer/access.lua#L488, w/o argumets that function returns 100 right, so the best solution i think to parametrize it through config

**Comment by ghost** [comment, marked as answer]
This is a known issue and how to fix it quickly: https://github.com/Kong/kong/issues/1862#issuecomment-264415914

The reason why the default limit is 100 is because of the limit in OpenResty, refer to: https://github.com/openresty/lua-nginx-module#ngxreqget_uri_args

I think Kong could probably provide a configuration item to control the removal of the default 100 args limit.

**Reply by dbelozerovx1** [reply]
thanks! there's another place where kong is limiting, means core as well as transform plugin does it, unfortunately there's no parameter for kong config to change it :(

**Reply by ghost** [reply]
You can create issue from this discussion, let's see what maintainers think about it.

**Reply by dbelozerovx1** [reply]
yeah, i think ill do it that way, thanks again!
